handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1109,7 +1122,9 @@ public final synchronized void closeOutbound() {
 
     @Override
     public final synchronized boolean isOutboundDone() {
-        return isOutboundDone;
+        // Check if there is anything left in the outbound buffer.
+        // We need to ensure we only call SSL.pendingWrittenBytesInBIO(...) if the engine was not destroyed yet.
+        return outboundClosed && (isDestroyed() || SSL.pendingWrittenBytesInBIO(networkBIO) == 0);
```

Comment:
brain fart ... I see `isDestroyed()` is necessary now. Below is an alternative to avoid a volatile read:

```java
return outboundClosed && (networkBIO  == 0 || SSL.pendingWrittenBytesInBIO(networkBIO) == 0)
```